brainscore_vision/model_helpers/brain_transformation/behavior.py

```diff
@@ -215,3 +219,92 @@ def labels_to_indices(self, labels):
                 indices.append(label2index[label])
             index2label = OrderedDict((index, label) for label, index in label2index.items())
             return indices, index2label
+
+
+class OddOneOut(BrainModel):
+    def __init__(self, identifier: str, activations_model, layer: Union[str, List[str]]):
+        """
+        :param identifier: a string to identify the model
+        :param activations_model: the model from which to retrieve representations for stimuli
+        :param layer: the single behavioral readout layer or a list of layers to read out of.
+        """
+        self._identifier = identifier
+        self.activations_model = activations_model
+        self.readout = make_list(layer)
+        self.current_task = BrainModel.Task.odd_one_out
+        self.similarity_measure = 'dot'
+
+    @property
+    def identifier(self):
+        return self._identifier
+
+    def start_task(self, task: BrainModel.Task):
+        assert task == BrainModel.Task.odd_one_out
+        self.current_task = task
+
+    def look_at(self, triplets, number_of_trials=1):
+        # Compute unique features and image_pathst
+        stimuli = triplets.drop_duplicates(subset=['stimulus_id'])
+        stimuli = stimuli.sort_values(by='stimulus_id')
+
+        image_paths = set(stimuli.stimulus_paths)
+        def _sort_by_number(path):
+            return int(path.stem)
+
+        sorted_paths = sorted(image_paths, key=_sort_by_number)
+        stimuli.stimulus_paths = sorted_paths
```

Comment:
Now if I call ``features = self.activations_model(stimuli, layers=self.readout)`` the assertion fails and some strange things are happening:
1. the paths are not sorted anymore.
```
(Pdb) stimulus_paths[:5]
['.brainio/stimulus_Hebart2023/112.jpg', '.brainio/stimulus_Hebart2023/1459.jpg', '.brainio/stimulus_Hebart2023/632.jpg', '.brainio/stimulus_Hebart2023/1278.jpg', '.brainio/stimulus_Hebart2023/1561.jpg']
```
2. The assembly looks like this:
```
(Pdb) assembly
<xarray.NeuroidAssembly (stimulus_path: 1854, neuroid: 4096)>
array([[0.        , 0.        , 0.        , ..., 0.        , 1.3678241 ,
        0.        ],
       [0.        , 0.        , 0.        , ..., 0.        , 1.6144679 ,
        0.        ],
       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ],
       ...,
       [0.59153354, 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ],
       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,
        3.7824514 ],
       [0.6981266 , 0.        , 0.        , ..., 0.        , 0.        ,
        0.        ]], dtype=float32)
Coordinates:
  * stimulus_path  (stimulus_path) <U56 '/Users/linussommer/.brainio/stimulus...
  * neuroid        (neuroid) MultiIndex
  - neuroid_num    (neuroid) int64 0 1 2 3 4 5 ... 4090 4091 4092 4093 4094 4095
  - model          (neuroid) object 'alexnet' 'alexnet' ... 'alexnet' 'alexnet'
  - layer          (neuroid) object 'classifier.5' ... 'classifier.5'
  - channel        (neuroid) int64 0 1 2 3 4 5 ... 4090 4091 4092 4093 4094 4095
  - channel_x      (neuroid) float64 nan nan nan nan nan ... nan nan nan nan nan
  - channel_y      (neuroid) float64 nan nan nan nan nan ... nan nan nan nan nan
  - neuroid_id     (neuroid) object 'alexnet.classifier.5.0' ... 'alexnet.cla...
  ```
The assembly paths are like this: 
```
(Pdb) assembly_paths[:5]
['.brainio/stimulus_Hebart2023/112.jpg', '.brainio/stimulus_Hebart2023/1459.jpg', '.brainio/stimulus_Hebart2023/632.jpg', '.brainio/stimulus_Hebart2023/1278.jpg', '.brainio/stimulus_Hebart2023/1561.jpg']
```
So they correspond to the stimulus_paths. But:
```
(Pdb) assert (np.array(assembly_paths) == np.array(stimulus_paths)).all()
*** AssertionError
```
If we look closer we see that:
```
(Pdb) np.where(np.array(assembly_paths) != np.array(stimulus_paths))
(array([ 508, 1586]),)
(Pdb) assembly_paths[508]
'.brainio/stimulus_Hebart2023/471.jpg'
(Pdb) stimulus_paths[508]
'.brainio/stimulus_Hebart2023/47.jpg'
(Pdb) assembly_paths[1586]
'.brainio/stimulus_Hebart2023/1441.jpg'
(Pdb) stimulus_paths[1586]
'.brainio/stimulus_Hebart2023/1440.jpg'
```
also
```
(Pdb) len(set(stimulus_paths))
1192
(Pdb) len(set(assembly_paths))
1190
```
which seems very bad.